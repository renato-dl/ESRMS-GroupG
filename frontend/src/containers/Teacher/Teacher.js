import React from 'react';
import { api } from '../../services/api';
import {Icon,List, Container,Button} from 'semantic-ui-react';
import './Teacher.scss';
import { NoData } from '../../components/NoData/NoData';

export class Teacher extends React.Component{
  state = {
    subjectsList:[]
  };

  async componentDidMount() {
    const {params} = this.props.match;
    const teacherID = window.location.href.substr(window.location.href.indexOf('/teacher/') + 9);
    localStorage.setItem('teacherID', teacherID);

    const response = await api.teacher.getTeacherSubjects(params.teacherID);
    if (response) {
      this.setState({ subjectsList: response.data })
    }
  }

  onSubjectClick = (subjectID,subjectName) => {
    //const {params} = this.props.match;
    this.props.history.push(`/teacher/subjects/${subjectID}/${subjectName}/topics`);
  };

  onSubjectGradeClick = (subjectID,subjectName)=>{
    this.props.history.push(`/teacher/subjects/${subjectID}/${subjectName}/TeacherGrade`);
  }

    render() {
      
    if(this.state.subjectsList.length){
      return (
          <Container className="contentContainer">
              <h3 className="contentHeader">
                <Icon name='braille'/>
                Teaching Plan
              </h3>

              <List relaxed>
                  {this.state.subjectsList.map((subject, index) =>
                    <List.Item className="myListItem" key={index}>
                        <List.Icon name='book' size='large' verticalAlign='middle' />
                        <List.Content>
                            <List.Header as='div' className="subjectListName">
                              {subject.subject.toUpperCase()}
                            </List.Header>

                            <List.Description as='div' className="subjectListClass">
                              Class: {subject.class}
                              <Button className='Topics' color='blue'  onClick={() => this.onSubjectClick(subject.subjectId,subject.subject)}>
                              Topics
                              </Button>
                              <Button className='Grades'  color='blue' onClick={()=>this.onSubjectGradeClick(subject.subjectId,subject.subject)}>
                              Grades
                              </Button>
                            </List.Description>
                        </List.Content>
                    </List.Item>
                  )}
              </List>
          </Container>
      );
    }
    return(
      <Container className="contentContainer">
        <h3 className="contentHeader">
          <Icon name='braille'/>
          Teaching Plan
        </h3>
        <NoData/>
      </Container>
    );
  }
}
