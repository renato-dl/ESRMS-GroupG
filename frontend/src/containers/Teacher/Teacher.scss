.item.myListItem{
    background-color: #fff;
    padding:17px!important;
    height: 70px;
    border:1px solid #c5c5c5;
    border-left:5px solid #008272;
    margin-bottom:10px;
    font-size:16px;
}
.item.myListItem:hover{
    background-color: #E3EBEE;
    border:0;
    
    border-left:5px solid #008272;
}