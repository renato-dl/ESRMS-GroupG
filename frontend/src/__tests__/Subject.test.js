import React from 'react';
import {Teacher} from '../containers/Teacher/Teacher';
import {render} from '@testing-library/react'
import {getRouterPropsForTest} from '../utils';

describe('Testing Subject component', () => {
  
  test('Test if component is rendered', () => {
    const component = render(
      <Teacher {...getRouterPropsForTest()} />
    );
  });

});
