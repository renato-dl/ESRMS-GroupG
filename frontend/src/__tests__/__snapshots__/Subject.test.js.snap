// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Subject component Testing snapshot 1`] = `undefined`;
